Fails to build on 4.7.0

Building on Arch 4.7.0-1

```
make -C src
make[1]: Entering directory '/home/tim/Projects/dattobd/src'
if [ ! -f kernel-config.h ] || tail -1 kernel-config.h | grep -qv '#endif'; then ./genconfig.sh 4.7.0-1-ARCH; fi;
generating configurations
make[2]: Entering directory '/home/tim/Projects/dattobd/src/configure-tests/feature-tests'
make[3]: Entering directory '/usr/lib/modules/4.7.0-1-ARCH/build'
make[3]: Leaving directory '/usr/lib/modules/4.7.0-1-ARCH/build'
make[2]: Leaving directory '/home/tim/Projects/dattobd/src/configure-tests/feature-tests'
performing configure test: HAVE_BDOPS_OPEN_INODE - not present
performing configure test: HAVE_BDOPS_OPEN_INT - not present
performing configure test: HAVE_BIO_BI_POOL - present
performing configure test: HAVE_BIO_BI_REMAINING - not present
performing configure test: HAVE_BIO_ENDIO_1 - present
performing configure test: HAVE_BIO_LIST - present
performing configure test: HAVE_BLKDEV_GET_BY_PATH - present
performing configure test: HAVE_BLKDEV_PUT_1 - not present
performing configure test: HAVE_BLK_QUEUE_MAX_SECTORS - not present
performing configure test: HAVE_BLK_SET_STACKING_LIMITS - present
performing configure test: HAVE_BVEC_ITER - present
performing configure test: HAVE___DENTRY_PATH - not present
performing configure test: HAVE_DENTRY_PATH_RAW - present
performing configure test: HAVE_D_UNLINKED - present
performing configure test: HAVE_FILE_INODE - present
performing configure test: HAVE_GENHD_FL_NO_PART_SCAN - present
performing configure test: HAVE_IOPS_FALLOCATE - not present
performing configure test: HAVE_KERN_PATH - present
performing configure test: HAVE_MAKE_REQUEST_FN_INT - not present
performing configure test: HAVE_MAKE_REQUEST_FN_VOID - not present
performing configure test: HAVE_MERGE_BVEC_FN - not present
performing configure test: HAVE_NOOP_LLSEEK - present
performing configure test: HAVE_NO_QUEUE_LIMITS - not present
performing configure test: HAVE_NOTIFY_CHANGE_2 - not present
performing configure test: HAVE_PART_NR_SECTS_READ - present
performing configure test: HAVE_QUEUE_LIMITS_MAX_HW_SEGMENTS - not present
performing configure test: HAVE_SB_START_WRITE - present
performing configure test: HAVE_SUBMIT_BIO_WAIT - present
performing configure test: HAVE_SYS_OLDUMOUNT - not present
performing configure test: HAVE_TASK_STRUCT_TASK_WORKS_CB_HEAD - present
performing configure test: HAVE_TASK_STRUCT_TASK_WORKS_HLIST - not present
performing configure test: HAVE_THAW_BDEV_INT - present
performing configure test: HAVE_VFS_FALLOCATE - present
performing configure test: HAVE_VFS_UNLINK_2 - not present
performing configure test: HAVE_VZALLOC - present
make[2]: Entering directory '/home/tim/Projects/dattobd/src/configure-tests/feature-tests'
make[3]: Entering directory '/usr/lib/modules/4.7.0-1-ARCH/build'
  CLEAN   /home/tim/Projects/dattobd/src/configure-tests/feature-tests/.tmp_versions
  CLEAN   /home/tim/Projects/dattobd/src/configure-tests/feature-tests/Module.symvers
make[3]: Leaving directory '/usr/lib/modules/4.7.0-1-ARCH/build'
make[2]: Leaving directory '/home/tim/Projects/dattobd/src/configure-tests/feature-tests'
performing sys_mount lookup
grep: /boot/System.map-4.7.0-1-ARCH: No such file or directory
performing sys_umount lookup
grep: /boot/System.map-4.7.0-1-ARCH: No such file or directory
performing sys_oldumount lookup
grep: /boot/System.map-4.7.0-1-ARCH: No such file or directory
make -I/usr/include -C /lib/modules/4.7.0-1-ARCH/build SUBDIRS=/home/tim/Projects/dattobd/src modules
make[2]: Entering directory '/usr/lib/modules/4.7.0-1-ARCH/build'
  CC [M]  /home/tim/Projects/dattobd/src/dattobd.o
/home/tim/Projects/dattobd/src/dattobd.c: In function ‘do_truncate2’:
/home/tim/Projects/dattobd/src/dattobd.c:808:29: error: ‘struct inode’ has no member named ‘i_mutex’; did you mean ‘i_mode’?
  mutex_lock(&dentry->d_inode->i_mutex);
                             ^~
/home/tim/Projects/dattobd/src/dattobd.c:815:31: error: ‘struct inode’ has no member named ‘i_mutex’; did you mean ‘i_mode’?
  mutex_unlock(&dentry->d_inode->i_mutex);
                               ^~
/home/tim/Projects/dattobd/src/dattobd.c: In function ‘__file_unlink’:
/home/tim/Projects/dattobd/src/dattobd.c:965:23: error: ‘struct inode’ has no member named ‘i_mutex’; did you mean ‘i_mode’?
  mutex_lock(&dir_inode->i_mutex);
                       ^~
/home/tim/Projects/dattobd/src/dattobd.c:979:25: error: ‘struct inode’ has no member named ‘i_mutex’; did you mean ‘i_mode’?
  mutex_unlock(&dir_inode->i_mutex);
                         ^~
make[3]: *** [scripts/Makefile.build:296: /home/tim/Projects/dattobd/src/dattobd.o] Error 1
make[2]: *** [Makefile:1457: _module_/home/tim/Projects/dattobd/src] Error 2
make[2]: Leaving directory '/usr/lib/modules/4.7.0-1-ARCH/build'
make[1]: *** [Makefile:9: default] Error 2
make[1]: Leaving directory '/home/tim/Projects/dattobd/src'
make: *** [Makefile:12: driver] Error 2
```

`struct inode`:  [4.6](http://lxr.free-electrons.com/source/include/linux/fs.h?v=4.6#L589), [4.7](http://lxr.free-electrons.com/source/include/linux/fs.h?v=4.7#L602)

--- a/src/configure-tests/feature-tests/inode_lock.c
+++ b/src/configure-tests/feature-tests/inode_lock.c
@@ -1,0 +1,16 @@
+/*
+    Copyright (C) 2015 Datto Inc.
+
+    This file is part of dattobd.
+
+    This program is free software; you can redistribute it and/or modify it
+    under the terms of the GNU General Public License version 2 as published
+    by the Free Software Foundation.
+*/
+
+#include "../../includes.h"
+
+static inline void dummy(void){
+	struct inode i;
+	inode_lock(&i);
+}

--- a/src/dattobd.c
+++ b/src/dattobd.c
@@ -227,6 +227,17 @@ static inline int dattobd_call_mrf(make_request_fn *fn, struct request_queue *q,
 	#define blk_set_stacking_limits(ql) blk_set_default_limits(ql)
 #endif
 
+#ifndef HAVE_INODE_LOCK
+//#if LINUX_VERSION_CODE < KERNEL_VERSION(4,5,0)
+static inline void inode_lock(struct inode *inode){
+    mutex_lock(&inode->i_mutex);
+}
+
+static inline void inode_unlock(struct inode *inode){
+    mutex_unlock(&inode->i_mutex);
+}
+#endif
+
 /*********************************MACRO/PARAMETER DEFINITIONS*******************************/
 
 //printing macros
@@ -805,14 +816,14 @@ static int do_truncate2(struct dentry *dentry, loff_t length, unsigned int time_
 	ret = should_remove_suid(dentry);
 	if(ret) newattrs.ia_valid |= ret | ATTR_FORCE;
 
-	mutex_lock(&dentry->d_inode->i_mutex);
+	inode_lock(dentry->d_inode);
 #ifdef HAVE_NOTIFY_CHANGE_2
 //#if LINUX_VERSION_CODE < KERNEL_VERSION(3,13,0)
 	ret = notify_change(dentry, &newattrs);
 #else
 	ret = notify_change(dentry, &newattrs, NULL);
 #endif
-	mutex_unlock(&dentry->d_inode->i_mutex);
+	inode_unlock(dentry->d_inode);
 
 	return ret;
 }
@@ -962,7 +973,6 @@ static int __file_unlink(struct file *filp, int close, int force){
 		LOG_ERROR(ret, "error getting write access to vfs mount");
 		goto file_unlink_mnt_error;
 	}
-	mutex_lock(&dir_inode->i_mutex);
 
 #ifdef HAVE_VFS_UNLINK_2
 //#if LINUX_VERSION_CODE < KERNEL_VERSION(3,13,0)
@@ -976,7 +986,6 @@ static int __file_unlink(struct file *filp, int close, int force){
 	}
 
 file_unlink_error:
-	mutex_unlock(&dir_inode->i_mutex);
 	mnt_drop_write(mnt);
 
 	if(close && (!ret || force)) file_close(filp);

